enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,172 @@
+---
+title: cli-in-cluster-management
+authors:
+  - "@sallyom"
+reviewers:
+  - TBD
+  - "@soltysh"
+approvers:
+  - TBD
+  - "@soltysh"
+creation-date: 2019-12-03
+last-updated: 2020-03-24-20
+status: provisional
+---
+
+# CLI In-Cluster Management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal is describing the mechanism for how CLI authors can deliver tools to OpenShift clusters.  An in-cluster feature
+is needed to manage various CLIs available for OpenShift and related services.  
+The goal is for a user to discover, install and upgrade tools easily and from a single
+location.  
+
+Each component is responsible for building and publishing its artifacts and registering information regarding supplied binaries.
+Currently, that location is [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/),
+and this makes it difficult for disconnected installations to mirror them.
+
+The CLI in-cluster manager will retrieve binaries from images that package a CLI's artifacts.  Through the use of local mirrored
+registries, disconnected environments will have access to OpenShift CLIs.  Currently, we provide disconnected environments 
+cli-artifacts and oc download links that do not require anything outside the cluster.  The goal is to provide the same for other
+binaries/tools/CLIs.
+
+The CLI in-cluster manager will provide a mechanism to provide, list, install, and upgrade OpenShift tools/plugins/CLIs from
+within a cluster.  The [ConsoleCLIDownload CRD](https://github.com/openshift/api/blob/master/console/v1/0000_10_consoleclidownload.crd.yaml) 
+will be extended to meet the requirements of a new CLI manager.  Each CLI will provide necessary information about its provided artifacts,
+including its description and location within the image.  
+
+An oc subcommand, `oc krew`,  will allow users to interact with CLI CRs to list, update, and install CLIs in the $HOME directory.
```

Comment:
Drop this sentence, you already said just that w/o naming `oc krew` but that's fine, few lines above.

Reply:
dropped